ndk/src/relay/pool/index.test.ts

```diff
@@ -3,8 +3,6 @@ import { NDK } from "../../ndk/index.js";
 import { NDKRelay } from "../index.js";
 import { NDKRelaySet } from "../sets/index.js";
 
-const ndk = new NDK();
-
 describe("NDKPool", () => {
```

Comment:
@pablof7z @verbiricha the tests in this file are flakey and intermittently throwing errors. Even though the tests in NDK are somewhat thin, I'd love to at least get to a passing setup.

Sometimes I get: 

```
node:internal/process/promises:289
            triggerUncaughtException(err, true /* fromPromise */);
            ^

[UnhandledPromiseRejection: This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). The promise rejected with the reason "undefined".] {
  code: 'ERR_UNHANDLED_REJECTION'
}
```

Other times I get this...

```
ReferenceError: You are trying to `import` a file after the Jest environment has been torn down. From src/relay/pool/index.test.ts.

      at Object.createBufferingLogger [as BufferingLogger] (../../../node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/utils.js:24:23)
      at new WebSocketConnection (../../../node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/WebSocketConnection.js:42:25)
      at WebSocketClient.Object.<anonymous>.WebSocketClient.succeedHandshake (../../../node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/WebSocketClient.js:343:22)
      at WebSocketClient.Object.<anonymous>.WebSocketClient.validateHandshake (../../../node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/WebSocketClient.js:332:10)
      at ClientRequest.handleRequestUpgrade (../../../node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/WebSocketClient.js:261:14)
/Users/jeff/src/listr.lol/node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/utils.js:24
    var logFunction = require('debug')(identifier);
                                      ^

TypeError: require(...) is not a function
    at Object.createBufferingLogger [as BufferingLogger] (/Users/jeff/src/listr.lol/node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/utils.js:24:39)
    at new WebSocketConnection (/Users/jeff/src/listr.lol/node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/WebSocketConnection.js:42:25)
    at WebSocketClient.Object.<anonymous>.WebSocketClient.succeedHandshake (/Users/jeff/src/listr.lol/node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/WebSocketClient.js:343:22)
    at WebSocketClient.Object.<anonymous>.WebSocketClient.validateHandshake (/Users/jeff/src/listr.lol/node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/WebSocketClient.js:332:10)
    at ClientRequest.handleRequestUpgrade (/Users/jeff/src/listr.lol/node_modules/.pnpm/websocket@1.0.34/node_modules/websocket/lib/WebSocketClient.js:261:14)
    at ClientRequest.emit (node:events:514:28)
    at TLSSocket.socketOnData (node:_http_client:575:11)
    at TLSSocket.emit (node:events:514:28)
    at addChunk (node:internal/streams/readable:343:12)
    at readableAddChunk (node:internal/streams/readable:316:9)
    at TLSSocket.Readable.push (node:internal/streams/readable:253:10)
    at TLSWrap.onStreamRead (node:internal/stream_base_commons:190:23)
```

Reply:
I think this is caused by lingering WebSocket connections, you can see if you run tests with `--detectOpenHandles`. I think we should add a `.disconnect` method to `NDK` to disconnect the websockets and clean up open file handles, then tests can use that to gracefully shut down. Thoughts @pablof7z ? I'm running into the same issue on a test suite that uses NDK.